pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -1695,3 +1695,192 @@ func createChainedEcOpHinter(resolver hintReferenceResolver) (hinter.Hinter, err
+			var xD4Big big.Int
+
+			xValues[0].BigInt(&xD3Big)
+			xValues[1].BigInt(&xD4Big)
+
+			base, ok := secp_utils.GetBaseBig()
+			if !ok {
+				return fmt.Errorf("getBaseBig failed")
+			}
+
+			//> p = pack(ids.P, PRIME)
+			pBig, err := secp_utils.SecPPacked(pValues)
+			if err != nil {
+				return err
+			}
+
+			//> x = pack(ids.x, PRIME) + as_int(ids.x.d3, PRIME) * ids.BASE ** 3 + as_int(ids.x.d4, PRIME) * ids.BASE ** 4
+			xPacked, err := secp_utils.SecPPacked(xValues)
+			if err != nil {
+				return err
+			}
+
+			base3Big := new(big.Int)
+
+			base3Big.Exp(&base, big.NewInt(3), big.NewInt(0))
+
+			base4Big := new(big.Int)
+
+			base4Big.Exp(&base, big.NewInt(4), big.NewInt(0))
+
+			xBig := new(big.Int)
+
+			xBig.Mul(&xD3Big, base3Big)
+
+			xBig.Add(xBig, xBig.Mul(&xD4Big, base4Big))
+
+			xBig.Add(xBig, &xPacked)
+
+			//> y = pack(ids.y, PRIME)
+			yBig, err := secp_utils.SecPPacked(yValues)
+			if err != nil {
+				return err
+			}
+
+			//> value = res = div_mod(x, y, p)
+			res, err := secp_utils.Divmod(xBig, &yBig, &pBig)
+			if err != nil {
+				return err
+			}
+
+			return ctx.ScopeManager.AssignVariables(map[string]any{"value": &res})
+		},
+	}
+}
+
+func createBigIntPackDivModHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
+	x, err := resolver.GetResOperander("x")
+	if err != nil {
+		return nil, err
+	}
+
+	y, err := resolver.GetResOperander("y")
+	if err != nil {
+		return nil, err
+	}
+
+	p, err := resolver.GetResOperander("P")
+	if err != nil {
+		return nil, err
+	}
+
+	return newBigIntPackDivModHint(x, y, p), nil
+}
+
+// SafeDiv hint safely divides two numbers and assigns the result based on a condition
+//
+// `newSafeDivHint` does not take any arguments
+//
+// `newSafeDivHint` assigns the result as `value` and sets `flag` based on the result in the current scope
+func newBigIntSafeDivHint() hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "SafeDiv",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> k = safe_div(res * y - x, p)
+			//> value = k if k > 0 else 0 - k
+			//> ids.flag = 1 if k > 0 else 0
+
+			x, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "x")
+			if err != nil {
+				return err
+			}
+
+			y, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "y")
+			if err != nil {
+				return err
+			}
+
+			p, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "p")
+			if err != nil {
+				return err
+			}
+
+			res, err := hinter.GetVariableAs[*big.Int](&ctx.ScopeManager, "res")
+			if err != nil {
+				return err
+			}
+
+			//> k = safe_div(res * y - x, p)
+			tmp := new(big.Int)
+			tmp.Mul(res, y)
+			tmp.Sub(tmp, x)
+			k := new(big.Int).Div(tmp, p)
+
+			//> value = k if k > 0 else 0 - k
+			value := new(big.Int).Abs(k)
+
+			//> ids.flag = 1 if k > 0 else 0
+			flag := big.NewInt(0)
+			if k.Sign() > 0 {
+				flag.SetInt64(1)
+			}
+
+			err = ctx.ScopeManager.AssignVariables(map[string]any{"value": value, "flag": flag})
```

Comment:
here, `flag` is not a scope variable. `ids` identifier in the hint means it is a variable  of the cairo vm, hence you need to write to memory instead of assigning in scope in that case